chef/cookbooks/updater/recipes/default.rb

```diff
@@ -40,6 +45,11 @@
       zypper_command += " --auto-agree-with-licenses"
     end
 
+    execute "refresh PTF repository" do
+      command "zypper --non-interactive --gpg-auto-import-keys refresh -fr PTF"
+      only_if { repos.keys.include? "PTF" }
```

Comment:
how likely is it that zypper exits with a non-zero exit code? this is not handled here, but should probably be ignored. also array style command execution might be preferred, although thats just a minor nit.